app/Http/Controllers/API/CategoryController.php

```diff
@@ -39,17 +39,20 @@ public function store(CategoryRequest $request): \Illuminate\Http\JsonResponse
             "categories" => new CategoryCollection($this->getCategoryInTree()) ,
         ] , 201);
     }
-
     /**
      * Update the specified resource in storage.
      *
      * @param CategoryRequest $request
      * @param \App\Models\Category $category
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\JsonResponse
      */
     public function update(CategoryRequest $request, Category $category)
     {
-        //
+        Category::update($request->validated());
+        $this->getCategoryInTree()->update($category,$request);
```

Comment:
What is this?
For what purpose did you do this?
If you do not have a specific goal, delete this line

Reply:
I meant that the tree should be updated based on category and request